go/general: comments for disapprove (#1223)
exercism/website-copy: automated-comments/go/general/avoid_printing_and_logging.md

Avoid printing and logging via `stdout` and `stderr`.

Usually it is not a good idea to output something to `stdout` or `stderr` in lower level packages.

> That is done in higher level or the `main` package, that have the necessary context to for example know what to do with an error.

Reviewer: ```suggestion
That is done in higher level or the `main` package, that have the necessary context to know what to do with an error.
```

> Usually it is not a good idea to output something to `stdout` or `stderr` in lower level packages.

Reviewer 2: ```suggestion
Avoid printing and logging via `stdout` and `stderr`.

Usually it is not a good idea to output something to `stdout` or `stderr` in lower level packages.
```

That is done in higher level or the `main` package, that have the necessary context to for example know what to do with an error.
Using `log.Println` or `fmt.Println` in a lower level package can not be overwritten from the outside.

> Best avoid them in the exercises here. They are useful for debugging but should be removed before submitting.

Reviewer 2: ```suggestion
Printing and logging are useful for debugging, but should be removed before submitting an exercise, or releasing a package.
```

Or something like that. The overal advice here is _great_, and you probably want to make sure it's clear that you basically never want it, unless you're writing an application (if I understood correctly).

Reviewer: I like this suggestion.

> It might be that you feel like an error should be returned by a function but there is no error return value in the function signature.

Reviewer 2: Can this be rephrased so there is no "you" or "feel"?

Author: ```suggestion
Sometimes it looks like the function signature should be changed to return an error.
```

> In the real world you could change this, but in these exercises the function signature is pre-determined.

Reviewer 2: Can this be rephrased so there is no "you"?

Author: ```suggestion
In contrast to the real world, the function signature is pre-determined in these exercises.
```

> Best discuss this with your mentor. An exception might not be necessary after all or the exercise might need to be adjusted.

Reviewer 2: The mentor most likely will get this comment, not the student. 

![image](https://user-images.githubusercontent.com/1964376/63133963-b5bd7400-bfc7-11e9-8f0b-96a50a148196.png)

So perhaps flip the wording?

Reviewer 3: I don't think we should use the word `exception` here, as it is ambiguous. Do we mean raising an exception (of which there are none in Go?) or do we mean an exception that proves the rule?

Author: ```suggestion
There's room for discussion: to return an error might not be necessary after all or the exercise might need to be adjusted.
```
